### PR TITLE
Builder for setting Action Policies for a probe

### PR DESCRIPTION
**Pull request by pallavidn**
Updated the SDK probe registration protobuf message builders with a builder for setting action policies. 
- Action policies are set by a probe. Each action policy is associated with an  EntityType and represents a list of actions with their modes (Disabled, Recommend, etc.).
- Create a default action policy for probes that do not support action executor.

### CODE REVIEW

File: pkg/probe/probe_registration.go [hunk 1]
```diff
@@ -109,10 +110,22 @@ func (dMetadata *DiscoveryMetadata) SetFullRediscoveryIntervalSeconds(fullDiscov
 
 // IActionPolicyProvider provides the policies for action types supported for
 // different entity types by the probe
+// Returns action policies defined by the probe. Each action policy is associated with an
+// EntityType and represents a list of actions with their modes (Disabled, Recommend, etc.).
 type IActionPolicyProvider interface {
 	GetActionPolicy() []*proto.ActionPolicyDTO
 }
 
+// The default action policy metadata if the probe does not support action executor.
```
Comment (songbinliu) [inline pkg/probe/probe_registration.go]: Any more detail about the purpose of this DefaultActionPolicyMetadata?
It defines `RecommondOnly` for action `ActionItemDTO_NONE` for entity `EntityDTO_UNKNOWN`.

File: pkg/probe/probe_registration.go [hunk 2]
```diff
@@ -84,7 +85,7 @@ func (dMetadata *DiscoveryMetadata) GetPerformanceRediscoveryIntervalSeconds() i
 func (dMetadata *DiscoveryMetadata) SetIncrementalRediscoveryIntervalSeconds(incrementalDiscovery int32) {
 	if incrementalDiscovery >= DEFAULT_MIN_DISCOVERY_IN_SECS {
 		dMetadata.incrementalDiscovery = incrementalDiscovery
-	} else {
+	} else if incrementalDiscovery != DISCOVERY_NOT_SUPPORTED {
```
[the remark this answers is not in the document]
Comment (songbinliu) [reply pkg/probe/probe_registration.go]: Also, if the input value is not used, shall we log the real value that is used?

**Comment by enlinxu** [comment]
Can we merge it?

**Comment by ading1977** [comment]
Close as this is out-dated.

**Pull request closed without merging**